### ISSUE TITLE
How can I configure the wait for emulator to boot timeout?

### ISSUE THREAD

**Issue by ZakTaccardi** [MarathonLabs/marathon#363]
I've set the `adbInitTimeout` to be 12 minutes (`12 * 60 * 1_000` in ms).

However I'm seeing the following exception after ~10 seconds when I attempt to run `/.gradlew :lib:marathonDebugAndroidTest` (with no emulators booted or booting).

```
com.malinskiy.marathon.exceptions.NoDevicesException: No devices found.
No devices found
```

The [`adbInit`](https://malinskiy.github.io/marathon/ven/android.html#adb-initialisation-timeout) docs are a little short on what ADB init means, and it would be useful to mention if it relates at all to the amount of time to wait for emulators to boot there.

**Comment by Malinskiy** [comment]
If there are no devices for 30 seconds, then marathon will throw `NoDevicesException`. This behaviour is not configurable at the moment

**Comment by Malinskiy** [comment]
The boot wait is a bit different to NoDevices state. If there is at least one device booted then the test execution will proceed. If there are no devices after the initial 30s period (e.g. all devices are still booting) then marathon will throw the exception as I described above.

**Comment by ZakTaccardi** [comment]
> This behaviour is not configurable at the moment

If this could be configurable, that would be awesome. I guess this issue can be retrofitted as a request to make this timeout configurable.

> If there is at least one device booted then the test execution will proceed. 

I'm booting multiple devices, so they come online one at a time. Is there any benefit to waiting for all emulators to boot before having marathon begin test execution?

> The boot wait is a bit different to NoDevices state.

Yeah, I basically just want to control the timeout for `NoDevicesException` to be thrown. As far as I understand it, I don't see a benefit in waiting for all emulators to complete booting before test execution to begin

**Comment by hram** [comment]
30 seconds is to short. If I want to build marathon with different timeout where can I change this value?

**Comment by Malinskiy** [comment]
I'd rather see a configuration PR than a custom fork with a different timeout value. It's also a bit more complicated since we now have 2 versions of vendor-android in the develop. The ddmlib implementation is not really configurable fully for proper timeouts, which is one of the reasons why we're building another vendor module in the first place.

For adam (new android vendor module) the logic for waiting for devices is [here](https://github.com/Malinskiy/marathon/blob/develop/vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamDeviceProvider.kt#L74).

Keep in mind that if you want to add this option you'd have to implement it for ddmlib also

**Comment by tagantroy** [comment]
Fixed in #389
